IronFrame/FirewallRuleSpec.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IronFrame
+{
+    public class FirewallRuleSpec
+    {
+        public Protocol Protocol { get; set; }
+
+        public List<IPRange> Networks { get; set; }
+
+        public List<PortRange> Ports { get; set; } 
+
+        public string RemoteAddresses
+        {
+            get
+            {
+                if (Networks == null)
+                {
+                    return "*";
+                }
+
+                return String.Join(",", Networks.Select(x => x.Start + "-" + x.End));
+            }
+        }
+
+        public string RemotePorts
+        {
+            get
+            {
+                if (Ports == null)
+                {
+                    return "*";
+                }
+
+                return String.Join(",", Ports.Select(x => x.Start + "-" + x.End));
+            }
+        }
+    }
+
+    public enum Protocol
+    {
+        All, Tcp, Udp
+    }
+
+    public class IPRange
+    {
+        public string Start { get; set; }
+        public string End { get; set; }
+    }
+
+    public class PortRange
+    {
+        public UInt16 Start { get; set; }
```

Comment:
The C# intrinsic type for `UInt16` is `ushort`.  But you should consider using `int` for public types and then casting to the smaller unsigned value at the lower level.  The issue is that not all .NET languages support unsigned types.  You'll see this pattern in the framework.

For example, for the System.Net.IPEndPoint type (https://msdn.microsoft.com/en-us/library/system.net.ipendpoint.port(v=vs.110).aspx) the `Port` property is an `int`.